app/gosns/gosns.go

```diff
@@ -485,6 +491,132 @@ func DeleteTopic(w http.ResponseWriter, req *http.Request) {
 
 }
 
+func PublishBatch(w http.ResponseWriter, req *http.Request) {
+	content := req.FormValue("ContentType")
+	topicArn := req.FormValue("TopicArn")
+
+	arnSegments := strings.Split(topicArn, ":")
+	topicName := arnSegments[len(arnSegments)-1]
+	topic, ok := app.SyncTopics.Topics[topicName]
+	if !ok {
+		createErrorResponse(w, req, "TopicNotFound")
+		return
+	}
+
+	batchMessageIdToMessageBody := make(map[string]string, 10)
+	batchMessageIdToMessageStructure := make(map[string]string, 10)
+	batchMessageIdToMessageAttributes := make(map[string]map[string]app.MessageAttributeValue, 10)
+	batchMessageIdToSubject := make(map[string]string, 10)
+
+	for memberIndex := 1; true; memberIndex++ {
+		if memberIndex > 10 {
+			createErrorResponse(w, req, "TooManyEntriesInBatchRequest")
+			return
+		}
+		thisMessageFormKey := "PublishBatchRequestEntries.member." + strconv.Itoa(memberIndex)
+
+		batchMessageId := req.FormValue(thisMessageFormKey + ".Id")
+		if batchMessageId == "" {
+			// This is a required field, its absence likely indicates there are no further entries.
+			// TODO: how to tell the difference in case this is validation failure?
+			break
+		}
+		if _, ok := batchMessageIdToMessageBody[batchMessageId]; ok {
+			createErrorResponse(w, req, "BatchEntryIdsNotDistinct")
+			return
+		}
+
+		thisMessageBody := req.FormValue(thisMessageFormKey + ".Message")
+		thisMessageStructure := req.FormValue(thisMessageFormKey + ".MessageStructure")
+		thisMessageSubject := req.FormValue(thisMessageFormKey + ".Subject")
+
+		thisMessageAttributes := make(map[string]app.MessageAttributeValue)
+
+		for i := 1; true; i++ {
```

Comment:
Depending on your thoughts on the comment above, would it be clearer to throw a quick comment in here on what were trying to do?  I know we're setting the MessageAttributes, but it may take folks (me 😅) some digging to figure that out with the names here.